lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -136,16 +136,24 @@ namespace Js
         Var newVarInstance = nullptr;
         if (callInfo.Flags & CallFlags_New)
         {
-          if (JavascriptProxy::Is(targetFunction))
-          {
-            JavascriptProxy* proxy = JavascriptProxy::FromVar(targetFunction);
-            Arguments proxyArgs(CallInfo(CallFlags_New, 1), &targetFunction);
-            args.Values[0] = newVarInstance = proxy->ConstructorTrap(proxyArgs, scriptContext, 0);
-          }
-          else
-          {
-            args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
-          }
+            if (!JavascriptProxy::Is(targetFunction))
+            {
+                // when target is not a proxy can make a new object in a "normal" way
+                args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
```

Comment:
I found another issue with this, but it's not due to your change: [https://github.com/Microsoft/ChakraCore/issues/5151](https://github.com/Microsoft/ChakraCore/issues/5151)

Reply:
Do you mind if I update this to fix #5151 as well? Seems silly to leave it as that will be an easy addition.

Reply:
Sure, you're welcome to if you like. Thanks!